### PR TITLE
Add support for UPM

### PR DESCRIPTION
**Pull request by loofou**
THis is a short pull request to give support for UPM (Unity Package Manager). I added a package.json with all the information I could find and updated all meta files. As meta-files are required for UPM to work, I removed the exception from the gitignore as well.

From what I've seen, this should be all that's required for UPM.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿{
+	"name": "space.exodrifter.unity-rumor",
+	"version": "5.0.0",
+	"description": "Rumor is a free, open source narrative content framework meant to be integrated into any Unity3D game that needs a dialog system. It is easy to diff, easy to read, and it is very flexible!",
+	"displayName": "Unity Rumor",
```
Comment (exodrifter) [inline package.json:5]: I think just "Rumor" would be fine here.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	"unity": "2020.3",
+	"author": {
+		"name": "exodrifter",
+		"email": "exodrifter@gmail.com",
```
Comment (exodrifter) [inline package.json:9]: I don't use this email address anymore, you should change it to `code@exodrifter.space`

**Review by exodrifter — COMMENTED** [review]
Thanks for opening a PR! :heart:

I'll be able to approve and merge this next week, when I will have the resources to test and make sure it works, but in the meantime I have a few suggestions.

Now that `.meta` files are no longer ignored, it would be good to add a `.gitattributes` file so git [does not generate diffs for `.meta` files](https://git-scm.com/docs/gitattributes#_marking_files_as_binary) and so github [does not include meta files in the  language statistics for the repository](https://github.com/github/linguist/blob/master/docs/overrides.md#generated-code)
```gitattributes
*.meta -diff linguist-generated
```

**Comment by loofou** [comment]
> Thanks for opening a PR! ❤️
> 
> I'll be able to approve and merge this next week, when I will have the resources to test and make sure it works, but in the meantime I have a few suggestions.
> 
> Now that `.meta` files are no longer ignored, it would be good to add a `.gitattributes` file so git [does not generate diffs for `.meta` files](https://git-scm.com/docs/gitattributes#_marking_files_as_binary) and so github [does not include meta files in the language statistics for the repository](https://github.com/github/linguist/blob/master/docs/overrides.md#generated-code)
> 
> ```gitattributes
> *.meta -diff linguist-generated
> ```

I actually didn't know about this. Why "lingust-generated" in particular? Wouldn't treating the meta file like a binary suffice?

File: package.json (hunk 2 above)
Comment (loofou) [reply package.json:9]: Yeah, sorry, I looked into some older projects of yours and through itch and your website to find those information :D

**Comment by exodrifter** [comment]
> I actually didn't know about this. Why "lingust-generated" in particular? Wouldn't treating the meta file like a binary suffice?

`linguist-generated` is a way of communicating with linguist so it knows that a file shouldn't be considered in the language statistics of the repository. It's the most specific and direct method available, so I wouldn't want to rely on the implementation of linguist happening to not consider binary files in the language statistics (I'm not sure if this is the case or not). There are also other benefits, such as GitHub showing a more specific message -- iirc it will say that the file is generated instead of saying it's a binary file while still allowing someone to view the diff if they would like.

**Review by exodrifter — APPROVED** [review]
Successfully tested on Linux with Unity 2021.3.4f1 Personal.

I wonder if UPM has some way to specify dependencies; adding the repository will result in compilation errors if the user doesn't have the Test Framework package installed (this package is normally installed by default in a new project).

Although, the tests are mostly meant for people working on this repository and not for end-users. Maybe there's some other way to not require the test framework to be installed when using it as an end-user.

**Comment by loofou** [comment]
> Successfully tested on Linux with Unity 2021.3.4f1 Personal.
> 
> I wonder if UPM has some way to specify dependencies; adding the repository will result in compilation errors if the user doesn't have the Test Framework package installed (this package is normally installed by default in a new project).
> 
> Although, the tests are mostly meant for people working on this repository and not for end-users. Maybe there's some other way to not require the test framework to be installed when using it as an end-user.

UPM supports dependencies, so you can add a dependecy field to the package.json and give it the UPM address for the unity testframework and it should make sure it's also installed. Didn't notice the dependency existed, sorry...

**Comment by exodrifter** [comment]
Fixed the test dependency issue in e3f62d6fb456e032887b3038635cd5ea05383272.

**Pull request merged**